src/main/java/util/Util.java

```diff
@@ -48,4 +51,43 @@ public static FluidStack[] toFluidStackArray(List<FluidStack> stacksList) {
         }
         return ret;
     }
+
+    /* If the number is less than 1, we round by the variable, otherwise to 2 */
+    public static String toPercentageFrom(BigInteger stored, BigInteger capacity, int round) {
+        BigDecimal dividend = new BigDecimal(stored);
+        BigDecimal divisor = new BigDecimal(capacity);
+
+        BigDecimal result = dividend.multiply(BigDecimal.valueOf(100)).divide(divisor, round, RoundingMode.HALF_UP);
+        if (result.compareTo(BigDecimal.ONE) > 0) {
+            result = result.setScale(2, RoundingMode.HALF_UP);
+        }
+        return result.toString();
+    }
+
+    /* Get a string like this: 4.56*10^25 */
+    public static String toStandardForm(BigInteger number) {
+        BigInteger abs = number.abs();
+        if (abs.compareTo(BigInteger.valueOf(Long.MAX_VALUE)) < 0) {
+            return toStandardForm(number.longValue());
+        }
+        String strNum = abs.toString();
+        int exponent = strNum.length() - 1;
+        return (number.signum() == -1 ? "-" : "") + strNum.charAt(0) + "." + strNum.substring(1, 3) + "*10^" + exponent;
+
+    }
+
+    /* Get a string like this: 4.56*10^5 */
+    public static String toStandardForm(long number) {
+        if (number == 0) {
+            return "0";
+        }
+
+        int exponent = (int) Math.floor(Math.log10(Math.abs(number)));
+        double mantissa = number / Math.pow(10, exponent);
+
+        // Round the mantissa to two decimal places
+        mantissa = Math.round(mantissa * 100.0) / 100.0;
+
+        return mantissa + "*10^" + exponent;
+    }
```

Comment:
Most of this could be replaced by using `java.lang.DecimalFormat`, which has built-in functions to do all of this and works with `BigDecimal` as well.

Quick example:

```java
        BigInteger billion = BigInteger.valueOf(1_000_000_000);
        BigInteger value = billion.multiply(billion).multiply(billion).multiply(BigInteger.valueOf(4567));
        BigInteger maxValue = value.multiply(BigInteger.valueOf(3));

        DecimalFormatSymbols dfs = new DecimalFormatSymbols(Locale.US);
        dfs.setExponentSeparator("x10^");
        DecimalFormat format = new DecimalFormat("0.00E0", dfs);
        System.out.println("Used Capacity: " + format.format(value) + " EU");
        // Prints "Used Capacity: 4.57x10^30 EU"

        DecimalFormat percent = new DecimalFormat("0.00%");
        System.out.println(percent.format(new BigDecimal(value).setScale(4, RoundingMode.HALF_UP)
                .divide(new BigDecimal(maxValue), RoundingMode.HALF_UP)));
        // Prints "33.33%"
```

Feel free to tailor the detailed specifications as needed.